airbyte-integrations/connectors/source-pardot/source_pardot/stream.py

```diff
@@ -42,29 +41,31 @@ def request_headers(
     def request_params(
         self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
     ) -> MutableMapping[str, Any]:
-        params = {
-            "format": "json",
-        }
-        start_date = self.config.get("start_date", None)
-        if start_date:
-            params.update({"created_after": pendulum.parse(start_date, strict=False).strftime(self.time_filter_template)})
-        if next_page_token:
+        schema = self.get_json_schema()
+        fields = ",".join(schema["properties"].keys())
+        params = {"fields": fields}
+        if next_page_token is not None:
             params.update(**next_page_token)
+        else:
+            start_date = self.config.get("start_date", None)
+            if start_date:
+                params.update({"createdAfter": pendulum.parse(start_date, strict=False).strftime(self.time_filter_template)})
+
+            params.update({"limit": self.limit})
         return params
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        results = response.json().get("result", {})
-        record_count = results.get("total_results")
+        results = response.json()
+        values = results.get("values", [])
         # The result may be a dict if one record is returned
-        if self.data_key in results and isinstance(results[self.data_key], dict):
-            yield results[self.data_key]
-        elif record_count and record_count > 0 and self.data_key in results:
-            yield from results[self.data_key]
+        if values is not None:
```

Comment:
It the comment above still accurate?  And can this ever be `None` now given that you have `[]` as the fallback?